File: 2002/101 VB.NET Samples/VB.NET - Data Access - Data Entry Form/ListItem.vb
Option Strict On
Option Explicit On 

'  This class is used to store the text and primary key of a record
'  that you want to list in list box.

Public Class ListItem
    Private mValue As String   ' Stores a named description of the item
    Private mID As Integer     ' Stores a primary key to the record

    Public Sub New(ByVal strValue As String, ByVal intID As Integer)
        mValue = strValue
        mID = intID
    End Sub

    Public Sub New()
        mValue = ""
        mID = 0
    End Sub

    Property ID() As Integer
        Get
            Return mID
        End Get
        Set(ByVal Value As Integer)
            mID = Value
        End Set
    End Property

    Property Value() As String
        Get
            Return mValue
        End Get
        Set(ByVal Value As String)
            mValue = Value
        End Set
    End Property

    ' When a list box displays an item in its collection
    ' it calls the ToString method to get the value to
    ' display so we will need to override this method
    ' so our class will return mValue.
    Public Overrides Function ToString() As String
        Return mValue
    End Function
End Class

File: 2003/Code Samples/Data Access Samples/Data Access - Data Entry Form/ListItem.cs

//  This class is used to store the text and primary key of a record
//  that you want to list in list box.
using System;


public class ListItem
{

    private string mValue;    //Stores a named description of the item;

    private int mID;      //Stores a primary key to the record;

    public ListItem(string strValue , int intID )
	{

        mValue = strValue;

        mID = intID;

    }

    public ListItem() 
	{

        mValue = string.Empty;
        mID = 0;
    }

    public int ID
	{

        get 
		{

            return mID;
        }

        set
		{
            mID = value;
        }

    }

    public string Value 
	{

        get 
		{

            return mValue;

        }

        set {

            mValue = value;

        }

    }

    // When a list box displays an item in its collection
    // it calls the Tostring method to get the value to
    // display so we will need to override this method
    // so our class will return mValue.

    public override string ToString()
	{

        return mValue;

    }

}

